### PR TITLE
add edit event endpoint

### PR DESCRIPTION
**Pull request by AnnPostnikova**
**1)** add edit event endpoint
**2)** add List of CreateCategoryRequest to CreateEventRequest
**ТЗ:** https://docs.google.com/document/d/1upUYu5MaGjBW2MKP-wHM2hwef8t6TPGzpVNkF8njJDI/edit#heading=h.7jjwan5qecem

### CODE REVIEW

File: src/EventService.Business/Commands/Event/CreateEventCommand.cs [hunk 1]
```diff
@@ -122,11 +140,24 @@ private async Task SendInviteEmailsAsync(List<Guid> userIds, string eventName)
...
     response.Body = await _eventRepository.CreateAsync(dbEvent);
 
-    await SendInviteEmailsAsync(dbEvent.Users.Select(x => x.UserId).ToList(), dbEvent.Name);
-
-    _contextAccessor.HttpContext.Response.StatusCode = response.Body is null
-      ? (int)HttpStatusCode.BadRequest
-      : (int)HttpStatusCode.Created;
+    await SendInviteEmailsAsync(dbEvent.Users.Select(x => x.UserId).ToList(), dbEvent.Name);
+
+    if(response.Body != null)
```
Comment (VLiamin) [inline src/EventService.Business/Commands/Event/CreateEventCommand.cs]: space

Comment (VLiamin) [reply src/EventService.Business/Commands/Event/CreateEventCommand.cs]: and not null

File: src/EventService.Validation/Event/CreateEventRequestValidator.cs [hunk 2]
```diff
@@ -60,13 +71,42 @@ await userService.CheckUsersExistenceAsync(users.Select(userRequest => userReque
...
+            countCategories = countCategories + ev.CategoriesRequests.Count();
+          }
+          else
+          {
+            countCategories = ev.CategoriesIds.Count();
+          };
+
+          return countCategories < 2;
+        })
+        .WithMessage("Count of categories to event must be no more than 1");
```
Comment (VLiamin) [inline src/EventService.Validation/Event/CreateEventRequestValidator.cs]: may be add dot?

File: src/EventService.Business/Commands/Event/CreateEventCommand.cs [hunk 3]
```diff
@@ -102,6 +110,16 @@ private async Task SendInviteEmailsAsync(List<Guid> userIds, string eventName)
         validationResult.Errors.ConvertAll(er => er.ErrorMessage));
     }
 
+    List<DbCategory> dbCategories = new();
+
+    if (!request.CategoriesRequests.IsNullOrEmpty())
+    {
+      foreach (CreateCategoryRequest categoryRequest in request.CategoriesRequests)
```
Comment (VLiamin) [inline src/EventService.Business/Commands/Event/CreateEventCommand.cs]: May be like this? 
image

File: src/EventService.Business/Commands/Event/CreateEventCommand.cs [hunk 4]
```diff
@@ -122,11 +140,24 @@ private async Task SendInviteEmailsAsync(List<Guid> userIds, string eventName)
...
-    _contextAccessor.HttpContext.Response.StatusCode = response.Body is null
-      ? (int)HttpStatusCode.BadRequest
-      : (int)HttpStatusCode.Created;
+    await SendInviteEmailsAsync(dbEvent.Users.Select(x => x.UserId).ToList(), dbEvent.Name);
+
+    if(response.Body != null)
+    {
+      if (dbCategories.Any())
+      {
+        foreach (DbCategory category in dbCategories)
```
Comment (VLiamin) [inline src/EventService.Business/Commands/Event/CreateEventCommand.cs]: May be add range instead of one category tp bd?

**Review by VLiamin — APPROVED** [review]

File: src/EventService.Data/Interfaces/IEventRepository.cs [hunk 5]
```diff
@@ -3,14 +3,17 @@
...
 
 [AutoInject]
 public interface IEventRepository
 {
-  Task<Guid?> CreateAsync(DbEvent dbEvent);
-  Task<bool> DoesExistAsync(Guid eventId);
+  Task<Guid?> CreateAsync(DbEvent dbEvent);
+  Task<bool> EditAsync(Guid eventId, JsonPatchDocument<DbEvent> request);
+  Task<bool> DoesExistAsync(Guid eventId, bool? isActive);
+  public Task<bool> IsEventCompletedAsync(Guid eventId);
```
Comment (SolovyovDmitry) [inline src/EventService.Data/Interfaces/IEventRepository.cs]: isn't public extra here?

File: src/EventService.Validation/Event/CreateEventRequestValidator.cs [hunk 6]
```diff
@@ -60,13 +71,42 @@ await userService.CheckUsersExistenceAsync(users.Select(userRequest => userReque
...
-        .WithMessage("Some of categories in the list doesn't exist.")
-        .Must(cat => cat.Count < 6)
-        .WithMessage("Count of categories to event must be no more than 5");
+        .WithMessage("Some of categories in the list doesn't exist.");
+      });
+
+      RuleFor(ev => ev)
+        .Must((ev) =>
+        {
+          int countCategories = default;
```
Comment (SolovyovDmitry) [inline src/EventService.Validation/Event/CreateEventRequestValidator.cs]: zero would be shorter)

File: src/EventService.Validation/EventCategory/CreateEventCategoryRequestValidator.cs [hunk 7]
```diff
@@ -13,7 +13,7 @@ public class CreateEventCategoryRequestValidator : AbstractValidator<CreateEvent
     IEventCategoryRepository eventCategoryRepository)
   {
     RuleFor(x => x.EventId)
-      .MustAsync(async (eventId, _) => await eventRepository.DoesExistAsync(eventId))
+      .MustAsync(async (eventId, _) => await eventRepository.DoesExistAsync(eventId, true))
```
Comment (SolovyovDmitry) [inline src/EventService.Validation/EventCategory/CreateEventCategoryRequestValidator.cs]: async/await not required

**Review by SolovyovDmitry — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**